pkg/proc/variables_test.go

```diff
@@ -2053,3 +2053,61 @@ func TestCapturedVarVisibleOnFirstLine(t *testing.T) {
 		}
 	})
 }
+
+func TestIssue4116(t *testing.T) {
+	protest.MustSupportFunctionCalls(t, testBackend)
+	protest.AllowRecording(t)
+
+	varTestcases := []varTest{
+		{"v.Model", true, "\"B\"", "", "string", nil},
+		{"v.A.Model", false, "main.(*A).Model", "", "func() string", nil},
+		{"v1.X", false, "main.(*B1).X", "", "func()", nil},
+		{"v1.A1.X", false, "0", "", "int", nil},
+		{"v2.X", false, "main.(*B2).X", "", "func()", nil},
+		{"v2.B2.X", false, "main.(*B2).X", "", "func()", nil},
+		{"v2.B2.A2.X", true, "0", "", "int", nil},
+		{"v2.A2.X", true, "0", "", "int", nil},
+		{"v7.X", false, "main.A7.X", "", "func()", nil},
+		{"x.X", false, "main.(*A7).X", "", "func()", nil},
+		{"x1.X", false, "", "", "", errors.New("x1 (type void) has no member X")},
+		{"x2.X", false, "main.(*A7).X", "", "func()", nil},
+		{"x3.X", false, "main.(*A7).X", "", "func()", nil},
+		{"v9.V.X", false, "main.(*B9).X", "", "func()", nil},
+		{"a.X", false, "main.A8.X", "", "func()", nil},
+		{"b.X", false, "", "", "", errors.New("b has no member X")},
+		{"c.X", false, "main.(*A8).X", "", "func()", nil},
+		{"d.X", true, "1", "", "int", nil},
+		{"v3.X", false, "main.(*B3).X", "", "func()", nil},
+		{"v3.B3.X", false, "main.(*B3).X", "", "func()", nil},
+		{"v4.X", false, "main.(*A4).X", "", "func()", nil},
+		{"v4.TestX.X", false, "main.(*A4).X", "", "func()", nil},
+		{"v5.X", true, "0", "", "int", nil},
+		{"v5.B5.X", false, "main.main.func1", "", "func() string", nil},
+		{"v5.B5.A5.X", false, "main.main.func1", "", "func() string", nil},
+		{"v5.A5.X", false, "main.main.func1", "", "func() string", nil},
+		{"*(ch.buf)", true, "[0]int []", "", "[0]int", nil},
+		{"*(v6.Chan.buf)", true, "[0]struct struct {} []", "", "[0]struct struct {}", nil},
+		{"*(v6.buf)", false, "", "", "", errors.New("v6 has no member buf")},
+	}
+
+	withTestProcess("issue4116", t, func(p *proc.Target, grp *proc.TargetGroup, fixture protest.Fixture) {
+		testCallFunctionSetBreakpoint(t, p, grp, fixture)
```

Comment:
This call doesn't do anything, the program will stop at `runtime.Breakpoint` anyway.